viz/MLSMapVisualization.cpp

```diff
@@ -123,27 +126,85 @@ struct DataHold : public MLSMapVisualization::Data
     }
 
     Eigen::Vector2d getResolution() const { return mls.getResolution(); }
-    void visualize(vizkit3d::PatchesGeode& geode) const
+    
+
+void visualize(vizkit3d::SurfaceGeode& geode) const
     {
-        //const GridMap<SPListST> &mls = *this;
         Vector2ui num_cell = mls.getNumCells();
-        for (size_t x = 0; x < num_cell.x(); x++)
-        {
-            for (size_t y = 0; y < num_cell.y(); y++)
+        double lastheight = 0;
+
+            for (size_t y = 0; y < num_cell.y()-1; y++) {
+                for (size_t x = 0; x < num_cell.x(); x++) {
+
+                    typedef typename MLSMap<Type>::CellType Cell;
+                    const Cell &list = mls.at(x, y);
+                    const Cell &nextlist = mls.at(x, y+1);
+
+                    if (list.size() && nextlist.size()){
+                        //geode.drawBox(p.getMean(), p.getHeight(), Vec3(p.getNormal()));
+                        auto patch = list.begin();
+                        auto neighborpatch = nextlist.begin();
+
+                        Eigen::Vector2d xypos = (maps::grid::Index(x, y).cast<double>() + maps::grid::Vector2d(0.5, 0.5)).array() * mls.getResolution().array();
+                        Eigen::Vector2d nxypos = (maps::grid::Index(x, y+1).cast<double>() + maps::grid::Vector2d(0.5, 0.5)).array() * mls.getResolution().array();
+
+                        float height = patch->getMax();
```

Comment:
Since the maximum value of each MLS patch is used, the optical visualization of the connected surfaces and the MLS patches will differ. Maximum is the mean + 1 sigma. Is this intended?

Reply:
isn't maximum the highest point? as it is the lowest patch, I need to use max to display walls correctly

Reply:
I checked it again, so max is actually equal to the mean. However I noticed a offset between the visualization of the sloped patches and the connected surface, but I would currently assume that this comes from the existing code.

Reply:
My impression was that offset comes from using the cell center and max from the lowest patch, rather than the 4 corner points with the actual hight. This also results in a offset where the surface begins at the borders (cell center instead of actual grid border)